lib/features/public_asset/presentation/public_asset_controller.dart

```diff
@@ -0,0 +1,159 @@
+import 'package:core/utils/platform_info.dart';
+import 'package:dartz/dartz.dart';
+import 'package:file_picker/file_picker.dart';
+import 'package:get/get.dart';
+import 'package:jmap_dart_client/jmap/account_id.dart';
+import 'package:jmap_dart_client/jmap/core/id.dart';
+import 'package:jmap_dart_client/jmap/core/session/session.dart';
+import 'package:jmap_dart_client/jmap/identities/identity.dart';
+import 'package:model/extensions/session_extension.dart';
+import 'package:tmail_ui_user/features/base/base_controller.dart';
+import 'package:tmail_ui_user/features/composer/domain/state/upload_attachment_state.dart';
+import 'package:tmail_ui_user/features/composer/domain/usecases/upload_attachment_interactor.dart';
+import 'package:tmail_ui_user/features/public_asset/domain/extensions/string_to_public_asset_extension.dart';
+import 'package:tmail_ui_user/features/public_asset/domain/state/create_public_asset_state.dart';
+import 'package:tmail_ui_user/features/public_asset/domain/usecase/create_public_asset_interactor.dart';
+import 'package:tmail_ui_user/features/identity_creator/presentation/identity_creator_controller.dart';
+import 'package:tmail_ui_user/features/identity_creator/presentation/utils/identity_creator_constants.dart';
+import 'package:tmail_ui_user/features/public_asset/presentation/model/public_asset_arguments.dart';
+import 'package:tmail_ui_user/features/upload/domain/extensions/platform_file_extension.dart';
+import 'package:tmail_ui_user/features/upload/domain/state/attachment_upload_state.dart';
+
+typedef PublicAssetId = Id;
+
+class PublicAssetController extends BaseController {
+  PublicAssetController(
+    this._uploadAttachmentInteractor,
+    this._createPublicAssetInteractor,
+    {required this.arguments}
+  );
+
+  final UploadAttachmentInteractor _uploadAttachmentInteractor;
+  final CreatePublicAssetInteractor _createPublicAssetInteractor;
+  final PublicAssetArguments? arguments;
+
+  List<PublicAssetId> oldPublicAssetIds = [];
+  List<PublicAssetId> newPublicAssetIds = [];
+  final _publicAssetStreamGroup = StreamGroup<Either<Failure, Success>>();
+  StreamSubscription<Either<Failure, Success>>? _publicAssetStreamSubscription;
+
+  StreamSubscription<Either<Failure, Success>>? get  publicAssetStreamSubscription
+    => _publicAssetStreamSubscription;
+  Session? get session => arguments?.session;
+  AccountId? get accountId => arguments?.accountId;
+  Identity? get identity => arguments?.identity;
+
+  void _handleUploadState(Either<Failure, Success> uploadState) {
+    log('PublicAssetController::handleUploadState::${uploadState.runtimeType}');
+    uploadState.fold(
+      (failure) {},
+      (success) {
+        log('PublicAssetController::handleUploadState::success::$success');
+        if (success is SuccessAttachmentUploadState) {
+          final filePath = success.fileInfo.filePath;
+          if (PlatformInfo.isMobile && filePath != null) {
+             _deleteCompressedFileOnMobile(filePath);
```

Comment:
Why not keep it util public asset to be created?

Reply:
PublicAsset only need the blobId of the file, not the file itself

Reply:
Ah, only compressed file in storage. thank you